### PR TITLE
[A5] Ensure a user can only get their own tickets

### PR DESCRIPTION
**Pull request by mbenford**

### CODE REVIEW

File: owasp-top10-2017-apps/a5/ecommerce-api/app/handlers/session.go [hunk 1]
```diff
@@ -33,6 +33,29 @@ func ReadCookie(c echo.Context) error {
...
+	cookie, err := c.Cookie("sessionIDa5")
+	if err != nil {
+		return "", err
+	}
+
+	token, err := jwt.Parse(cookie.Value, func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		}
+		return []byte("secret"), nil
```
Comment (rafaveira3) [inline owasp-top10-2017-apps/a5/ecommerce-api/app/handlers/session.go]: Ops, is that a hardcoded password on your code? 😅

Comment (mbenford) [reply owasp-top10-2017-apps/a5/ecommerce-api/app/handlers/session.go]: You tell me: https://github.com/globocom/secDevLabs/blob/master/owasp-top10-2017-apps/a5/ecommerce-api/app/handlers/session.go#L73 :sweat_smile:

Comment (rafaveira3) [reply owasp-top10-2017-apps/a5/ecommerce-api/app/handlers/session.go]: 🐰Easter Egg Found! 👏🏻

**Comment by rafaveira3** [comment]
Nice, @mbenford ! Your new code mitigates this vulnerability! 👏🏻

**Pull request closed without merging**